### ISSUE TITLE
RVC / Onnx usage, lack of information + lack of documentation.

### ISSUE THREAD

**Issue by codename0og** [axinc-ai/ailia-models#1341]
The process of obtaining prototxt isn't clarified and provided Onnx2prototxt seems to not function correctly, as in:
Inferences done on such exports are faulty.:

```
ailia.core.AiliaInvalidLayerException: code: -10 (Incorrect layer parameter. [broken or unsupported AI model file])
+ error detail : Layer:/enc_p/emb_phone/MatMul_output_0(MatMul_DNN) Error:Number of columns of first matrix and number of rows of second matrix are not same.
```

Generally, I request proper documentation or at least explanation on the requirements and the workflow.
As of now, there are certain onnx-exporting problems in rvc so it'd be nice to hear:
1. An overview of confirmed to work pth->onnx method for rvc's V2;
- which release of rvc was used to get onnx exports
- if not ^  what script ( in case it's custom ) was used.
- which onnx version + opset was used / is compatible; Yes, I am aware of 13/14 operators but it's been observed to be quite messy across few recent releases, compared to rvc v2 0618 beta release package
2. Proper explanation on obtaining the prototxt ( and again, if custom, better doc on that ) 

Thanks in advance

**Comment by kyakuno** [comment]
If you want to use rvc v2, please use ‘—version 2’ option for onnx inference.
You can use official onnx converter of rvc webui.
We are supporing with/without f0, and rvc v1/v2.
You can get onnx2prototxt.py from ailia SDK pacakage.
ailia SDK 1.2.16 supports opset <=17.

**Comment by kyakuno** [comment]
Also you can omit prototxt file by set None instead “xxx.onnx.prototxt” to ailia.Net constructor.

**Comment by kyakuno** [comment]
RVC version : 58370b048c580a34d33be4fde402bf847a9e3622 2023年7月26日 19:53:31

Training settings:
![train1](https://github.com/axinc-ai/ailia-models/assets/38881041/72b9b54c-8298-430b-9e5a-1d182f011e58)
![train2](https://github.com/axinc-ai/ailia-models/assets/38881041/968538ce-ee16-4920-974f-be4059f892db)

ONNX conversion:
![onnx](https://github.com/axinc-ai/ailia-models/assets/38881041/e7a0007c-d510-48df-934a-56fa2dfdfe73)

Inference : 

```
python3 rvc.py -m input.onnx --f0_method crepe_tiny --f0 1 --tgt_sr 48000 --version 2 
```

**Comment by codename0og** [comment]
> If you want to use rvc v2, please use ‘—version 2’ option for onnx inference. You can use official onnx converter of rvc webui. We are supporing with/without f0, and rvc v1/v2. You can get onnx2prototxt.py from ailia SDK pacakage. ailia SDK 1.2.16 supports opset <=17.

Well tbf, it's pointless, moving over onnx anyway. The current exporting method isn't dynamic, it's static and honestly, for that little of speed boost, losing accuracy / quality is def for a minus. For true potential of pytorch -> onnx in terms of rvc, we'd need dynamo exporting, sadly devs don't really care much about further rvc development, so it seems recently. Tho, thanks for info anyway, might come in handy in future, maybe.
